### ISSUE TITLE
[BUG] Error when using SRT in C++17 projects.

### ISSUE THREAD

**Issue by andersc** [Haivision/srt#1026]
When using libsrt.a in a C++17 (and versions above) project the compiler fails with the message -> 

> 'deprecated' attribute cannot be applied to types

in the header file srt.h

`In file included from /Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/main.cpp:15:
In file included from /Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/SRTNet.h:17:
/Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/../srt/srtcore/srt.h:198:9: error: 'deprecated' attribute cannot be applied to types
typedef SRT_ATR_DEPRECATED SRT_SOCKOPT SRT_SOCKOPT_DEPRECATED;
        ^
/Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/../srt/srtcore/srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
#define SRT_ATR_DEPRECATED [[deprecated]]
                             ^
In file included from /Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/SRTNet.cpp:5:
In file included from /Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/SRTNet.h:17:
/Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/../srt/srtcore/srt.h:198:9: error: 'deprecated' attribute cannot be applied to types
typedef SRT_ATR_DEPRECATED SRT_SOCKOPT SRT_SOCKOPT_DEPRECATED;
        ^
/Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper/../srt/srtcore/srt.h:84:30: note: expanded from macro 'SRT_ATR_DEPRECATED'
#define SRT_ATR_DEPRECATED [[deprecated]]`

To reproduce the Error create a C++17 project and use SRT

System MacOS Catalina -> 
10.15.2 (19C57)

Xcode information ->
Version 11.3 (11C29)

Cmake output->

"set(CMAKE_CXX_STANDARD 17)"

```
Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake -DCMAKE_BUILD_TYPE=Release -G "CodeBlocks - Unix Makefiles" /Users/anderscedronius/git/cppSRTWrapper/cppSRTWrapper
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
```

SRT Version ->
commit 51782ac338c9e680f4a18e3a52f9e707430caeb6 (HEAD -> master, origin/master, origin/HEAD)
Author: Mikołaj Małecki <mmalecki@haivision.com>
Date:   Mon Dec 16 13:18:09 2019 +0100
Added fixes for assignment copy compliance

My workaround for now was to comment out line 198 in srt.h
(//typedef SRT_ATR_DEPRECATED SRT_SOCKOPT SRT_SOCKOPT_DEPRECATED;)

C++14 does not produce the above errors. Lower than C++14 is not tested by me.

**Comment by ethouris** [comment]
Could you help me a little bit? I may have temporary problems with getting into this compiler version quickly; would you be able to make the following change at the definition of `SRT_DEPRECATED_OPTION` (around line 198 in srt.h)?

```
#define SRT_DEPRECATED_OPTION(value) ((SRT_SOCKOPT SRT_ATR_DEPRECATED)value)
```

The problem is that I can't repro this on any other compiler - at least to the grade where it results in a build break.

**Comment by andersc** [comment]
If I understod you correct I should change to as pictured ->

![image](https://user-images.githubusercontent.com/14168229/71122799-ace0e700-21e1-11ea-89af-d008cdbfa3de.png)

That works for me.

**Comment by ethouris** [comment]
Ok, I changed this a bit in the proposed fix in #1027 (every compiler behaves differently here). Nice if you could review.
